model/task/task.go

```diff
@@ -2788,6 +2790,31 @@ func (t *Task) SetResetWhenFinished() error {
 	)
 }
 
+// SetResetWhenFinishedWithInc requests that a task (that was marked to
+// automatically reset when finished via the agent status server) reset itself
+// when finished. It will also increment the number of automatic resets the task
+// has performed.
+func (t *Task) SetResetWhenFinishedWithInc() error {
+	if t.ResetWhenFinished {
+		return nil
+	}
+	t.ResetWhenFinished = true
+	t.NumAutomaticResets = t.NumAutomaticResets + 1
```

Comment:
nit: these should usually go after the update since the update could error and allow later logic involving this task to act on bad data.